### ISSUE TITLE
[Feature request] Annotations be aware of tag values in graphs

### ISSUE THREAD

**Issue by CarusDave** [grafana/grafana#6867]
Please include this information:
- What Grafana version are you using? 4.0.0-1478870419beta1
- What datasource are you using? InfluxDB 1.2.0~n201611110800
- What OS are you running grafana on? Debian 8.4

It would be useful if annotations could be tag aware when showing in graphs and not just show up on all.  Is there already a way to achieve this?

In this example we shouldn't see the host B tagged annotations on host A's graph.

![grafana1c](https://cloud.githubusercontent.com/assets/20041843/20967635/5c888d98-bc78-11e6-88e8-344b31455c92.jpg)
![grafana2c](https://cloud.githubusercontent.com/assets/20041843/20967648/6989f630-bc78-11e6-8117-292e96b26359.jpg)
![grafana3](https://cloud.githubusercontent.com/assets/20041843/20967652/6ea7dde4-bc78-11e6-9e15-78030b704ab6.jpg)

**Comment by torkelo** [comment]
you mean you want to filter annotations based on series shown on panel? How would you configure this?

**Comment by CarusDave** [comment]
Yes, that was what I was thinking. Not sure of the feasibility or overhead involved. To configure it, I'd thought you could select which annotations would be "matched" in the Graph editor, this would then compare the values returned by annotation "Tags" against any tag values found in the series for selected graphs. If they match, then the annotation is shown.

**Comment by mtanda** [comment]
It is similar my past feature request.
https://github.com/grafana/grafana/pull/6167
